### PR TITLE
The initial version.

### PR DESCRIPTION
**Pull request by Ayden-Franklin**
The skeleton project is built, which includes an API to obtain an identity. To satisfy this API, the information of a client should be posted to track the contributes and performance in the future.

### CODE REVIEW

File: api/controllers/v1/identities.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package api
+
+import (
+	"github.com/bwmarrin/snowflake"
+	"github.com/gin-gonic/gin"
+	"net/http"
+	database "tasks-dispatcher/internal/durable"
+)
+
+func generate() int64{
```
Comment (zhangjianye) [inline api/controllers/v1/identities.go:10]: I think it's better to move the generate function to a package named for example common or utilities

File: main.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package main
+
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+	"tasks-dispatcher/api/controllers/v1"
+)
+
+func main() {
+	 router := gin.Default()
```
Comment (zhangjianye) [inline main.go:10]: I think it's better to move the route code to another package

**Review by zhangjianye — CHANGES_REQUESTED** [review]
Is it better to encapsulate the response code like `c.Json()` to a common package?
and there are some MongoDB libraries for selection, such as mgo.v2 which seems has not been maintained, and qmgo opened by qiniu, we can make some comparison.

File: main.go (hunk 2 above)
Comment (Ayden-Franklin) [reply main.go:10]: It makes sense. I was going to contrive the project structure following the [guideline](https://github.com/golang-standards/project-layout). I'll append this change although I thought I would create a new issue to re-design before your suggestions.

**Comment by Ayden-Franklin** [comment]
> Is it better to encapsulate the response code like `c.Json()` to a common package?
> 
> and there are some MongoDB libraries for selection, such as mgo.v2 which seems has not been maintained, and qmgo opened by qiniu, we can make some comparison.

Let's consider them later.

File: api/controllers/v1/identities.go (hunk 1 above)
Comment (Ayden-Franklin) [reply api/controllers/v1/identities.go]: Good suggestion!

File: internal/apiserver/api/controllers/routers.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package api
```
Comment (zhangjianye) [inline internal/apiserver/api/controllers/routers.go:1]: I suggest you put this file in another package as a sibling of controllers

**Review by zhangjianye — CHANGES_REQUESTED** [review]
excellent, except for a file would better move to another package.

File: internal/apiserver/api/controllers/routers.go (hunk 3 above)
Comment (Ayden-Franklin) [reply internal/apiserver/api/controllers/routers.go:1]: Could you explain in more detail?

Comment (zhangjianye) [reply internal/apiserver/api/controllers/routers.go:1]: In my past experience, routers do not belong to controllers, they called controllers instead.
So I suggest put routers.go in a package named route which out of controllers

Comment (Ayden-Franklin) [reply internal/apiserver/api/controllers/routers.go:1]: It makes sense. However, routers are relevant to controllers that all belong to the API. I'd prefer to keep them in the same package `api`. What do you think?

**Review by zhangjianye — APPROVED** [review]

**Pull request merged**